### PR TITLE
P3 - Menu Item Page

### PR DESCRIPTION
**Pull request by djoyal**
This PR adds additional functionality and components to the P3 page. Specifically:

* The appearance of the `Good Reasons` component has been improved
* A new `Ingredients` section has been added that shows which modifiers have been selected
* The behaviour of the `Modifiers` modal has been improved to
   * Have a local state, so if you `X` out the changes you made are undone
   * Only save the state when you press `Save`
* Calories and Pricing for the menu item are now handled dynamically, so that they change when editing the modifiers
* Added some i18N entries related to the P3 page

### CODE REVIEW

File: packages/web/lib/theme.tsx [hunk 1]
```diff
@@ -73,19 +73,39 @@ const theme = {
...
+      },
+      extend: {
+        fontFamily: "FieldGrotWide",
+        borderRadius: "8px",
+      },
+      padding: {
+        horizontal: "16px",
+        vertical: "8px",
+      },
+    },
```
Comment (jfplataroti) [inline packages/web/lib/theme.tsx:109]: that's pretty cool!

Just to know, can we create any custom button with its own name? or it has to be the predefined names they provide for the buttons?

File: packages/web/lib/helpers/calculateItemCalories.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    modifierGroups
+      .flatMap(({ modifiers }) =>
+        modifiers.flatMap(({ calorieCount, selected }) => {
+          return Number.parseInt(calorieCount) * Number.parseInt(selected);
+        })
+      )
+      .filter((calories) => calories > 0)
+      .reduce((calories, total) => calories + total, 0)
+  );
+}
```
Comment (jfplataroti) [inline packages/web/lib/helpers/calculateItemCalories.ts:13]: 👍

File: packages/web/lib/theme.tsx [hunk 3]
```diff
@@ -1,12 +1,12 @@
 import { grommet } from "grommet/themes";
 import { deepMerge } from "grommet/utils";
-import { colors } from "grommet/themes/base";
```
Comment (jfplataroti) [inline packages/web/lib/theme.tsx:3]: thanks hehe

File: packages/web/components/selectedModifiers.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    <Box margin={{ bottom: "large", horizontal: "medium" }}>
+      <Box border="horizontal" margin={{ bottom: "medium" }}>
+        <Heading level={2} margin="large">
+          {t("ingredients")}
+        </Heading>
+      </Box>
+      <Grid columns={{ count: "fill", size: "xsmall" }}>
+        {modifierGroups.map(({ modifiers }) => {
+          return modifiers
+            .filter((modifier) => modifier.selected > 0)
```
Comment (jfplataroti) [inline packages/web/components/selectedModifiers.tsx:18]: not sure if we are going to need it, but if the modifierGroups it's a big iteration we should move the map and filter into a `useMemo` so we only calculate it when it changes.

File: packages/web/components/modifiers.tsx [hunk 5]
```diff
@@ -2,14 +2,26 @@ import { Box, Button, Card, Grid, Heading, Layer, Text } from "grommet";
...
   modifierGroups = [],
-  calories = 0,
-  onModifierClicked = (modifier, groupId) => {},
+  baseCalories = 0,
+  onSave = (updatedModifiers) => {},
   setVisible = (isVisible) => {},
 }) {
   const { t } = useTranslation("common");
+  const [modifiers, setModifiers] = useState(
+    JSON.parse(JSON.stringify(modifierGroups)) // A+ Method for Deep Cloning Objects
```
Comment (jfplataroti) [inline packages/web/components/modifiers.tsx:17]: nice! maybe we should have this function as a helper, like `deepClone` to be able to reuse it. :)

File: packages/web/lib/theme.tsx (hunk 1 above)
Comment (djoyal) [reply packages/web/lib/theme.tsx:109]: It looks like they only use predefined names: `primary`, `secondary` and `default`. Having the ability to add others would be very cool though!

I think if we're going to have more than those we either have to:
* Create a wrapper around `Button`
* Use the `default` style and just override it "in-place" (E.G. `<Button style={{ ... }} />`)

File: packages/web/components/selectedModifiers.tsx (hunk 4 above)
Comment (djoyal) [reply packages/web/components/selectedModifiers.tsx:18]: That's a great suggestion. I always forget about `useMemo`!

**Review by jfplataroti — APPROVED** [review]
Hey it looks great!! is nice to see the site with more and more features in each PR :)!!

I added some comments but are only thoughts.

**Pull request merged**